### PR TITLE
agent: get cody enabled status/version

### PR DESCRIPTION
**Pull request by SuperAuguste**
## Context

We need to be able to verify whether or not the instance has Cody enabled.

## Test Plan

Verified locally + all tests pass locally and on CI.

### CODE REVIEW

File: agent/src/protocol.ts [hunk 1]
```diff
@@ -95,6 +95,8 @@ export interface ClientCapabilities {
 
 export interface ServerInfo {
     name: string
+    isCodyEnabled: boolean
+    version: string
```
Comment (tjdevries) [inline agent/src/protocol.ts:99]: should be "codyVersion". At that point, i think `codyEnabled` and `codyVersion` look better

Comment (SuperAuguste) [reply agent/src/protocol.ts:99]: Good call, makes it clear that it's not the agent protocol version or anything like that :)

File: lib/shared/src/chat/client.ts [hunk 2]
```diff
@@ -169,5 +170,8 @@ export async function createClient({
             sendTranscript()
         },
         codebaseContext,
+        isCodyEnabled() {
```
Comment (tjdevries) [inline lib/shared/src/chat/client.ts:173]: but if cody isn't enabled, we do stuff above that doesn't make sense to run (like loading embeddings, etc)

Comment (SuperAuguste) [reply lib/shared/src/chat/client.ts:173]: Good call, rewriting this now

**Review by pjlast — APPROVED** [review]

**Pull request merged**